node/src/components/contract_runtime/operations.rs

```diff
@@ -99,3 +199,52 @@ pub(super) async fn execute(
     trace!(?result, "execute result");
     result
 }
+
+fn commit_step(
+    engine_state: &EngineState<LmdbGlobalState>,
+    metrics: &ContractRuntimeMetrics,
+    protocol_version: ProtocolVersion,
+    pre_state_root_hash: Digest,
+    era_report: &EraReport<PublicKey>,
+    era_end_timestamp_millis: u64,
+    next_era_id: EraId,
+) -> Result<StepSuccess, StepError> {
+    // Extract the rewards and the inactive validators if this is a switch block
+    let EraReport {
+        // Note: Highway does not slash, do nothing with the equivocators
+        equivocators: _,
+        rewards,
+        inactive_validators,
+    } = era_report;
+
+    let reward_items = rewards
+        .clone()
+        .into_iter()
+        .map(|(vid, value)| RewardItem::new(vid, value))
+        .collect();
+    let evict_items = inactive_validators
+        .clone()
+        .into_iter()
+        .map(EvictItem::new)
+        .collect();
+
+    let step_request = StepRequest {
+        pre_state_hash: pre_state_root_hash.into(),
+        protocol_version,
+        reward_items,
+        // Note: Highway does not slash; but another consensus protocol (e.g., BABE) could
```

Comment:
Whether to slash or not is not determined by the consensus protocol. (Also above.)
```suggestion
        // Note: The Casper Network does not slash, but another network could
```